### PR TITLE
SOLR-16542: Remove OverseerConfigSetMessageHandler; always do distributed mode

### PR DESCRIPTION
**Pull request by vinayakphegde**
https://issues.apache.org/jira/browse/SOLR-16542

# Description

remove the Overseer based processing of ConfigSets, thus always use DistributedCollectionConfigSetCommandRunner instead for ConfigSet processing

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

Additionally, I've updated some of the documentation to reflect these changes, but haven't made significant modifications to the overseer.doc file yet. I'll make sure to revise it once I'm confident that the changes are valid and functional.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java [hunk 1]
```diff
@@ -73,6 +69,10 @@ public class OverseerCollectionMessageHandler implements OverseerMessageHandler,
   Stats stats;
   TimeSource timeSource;
 
+  public interface Lock {
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java:72]: Perhaps this better belongs in LockTree?

File: solr/core/src/java/org/apache/solr/cloud/api/collections/DistributedCollectionConfigSetCommandRunner.java [hunk 2]
```diff
@@ -164,24 +163,7 @@ public void deleteAllAsyncIds() throws Exception {
...
-   *       queue, grabs the corresponding locks (write lock on the config set target of the API
-   *       command and a read lock on the base config set if any - the case for config set creation)
-   *       then executes the command using an executor service (it also checks the asyncId if any is
-   *       specified but async calls are not supported for Config Set API calls).
-   *   <li>In {@link org.apache.solr.cloud.OverseerTaskProcessor}.{@code Runner.run()} (run on an
-   *       executor thread) a call is made to {@link
-   *       org.apache.solr.cloud.OverseerConfigSetMessageHandler#processMessage} which does a few
-   *       checks and calls the appropriate Config Set method.
-   * </ul>
+   * to overseer queue and instead calls this method.
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/cloud/api/collections/DistributedCollectionConfigSetCommandRunner.java:166]: The former text was useful, I'd prefer it stay to show why it does what it does (due to legacy Overseer operation).

File: solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java [hunk 3]
```diff
@@ -129,17 +128,19 @@ public String toString() {
 
   private final Object waitLock = new Object();
 
-  protected OverseerMessageHandlerSelector selector;
-
   private OverseerNodePrioritizer prioritizer;
 
   private String thisNode;
 
+  private OverseerCollectionMessageHandler overseerCollectionMessageHandler;
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:135]: I suggest simply calling this "messageHandler" as it isn't ambiguous.

File: solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java [hunk 4]
```diff
@@ -149,7 +150,15 @@ public OverseerTaskProcessor(
     this.zkStateReader = zkStateReader;
     this.myId = myId;
     this.stats = stats;
-    this.selector = selector;
+    this.overseerCollectionMessageHandler =
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:153]: This needs to be closed.  The test failures are related to this; threads leak from it thus it wasn't closed.

Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:153]: or simply messageHandler as it isn't ambiguous?

File: solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java [hunk 5]
```diff
@@ -370,11 +380,12 @@ public void run() {
             if (log.isDebugEnabled()) {
               log.debug(
                   "{}: Get the message id: {} message: {}",
-                  messageHandler.getName(),
+                  this.overseerCollectionMessageHandler.getName(),
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:383]: previously, this referred to simply "messageHandler"; can we keep that?

File: solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java (hunk 1 above)
Comment (vinayakphegde) [reply solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java:72]: Okay, will change

File: solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java (hunk 3 above)
Comment (vinayakphegde) [reply solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:135]: Okay, will change that.

File: solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java (hunk 4 above)
Comment (vinayakphegde) [reply solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java:153]: Okay, got it. Thanks

**Pull request closed without merging**